app.py

```diff
@@ -25,34 +26,37 @@
 
 from mmif.utils import text_document_helper
 
+
 class TextSlicer(ClamsApp):
 
     def __init__(self):
         super().__init__()
 
     def _appmetadata(self):
-        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._load_appmetadata
-        # Also check out ``metadata.py`` in this directory.
-        # When using the ``metadata.py`` leave this do-nothing "pass" method here.
         pass
 
     def _annotate(self, mmif: Mmif, **parameters) -> Mmif:
-        for tf_view in mmif.get_all_views_contain(AnnotationTypes.TimeFrame):
-            if Uri.TOKEN in tf_view.metadata.contains:
-                asr_vid = tf_view.id
-                break
-        sliced = []
-        for ann in mmif.get_annotations_between_time(30000, 90000, time_unit="milliseconds"):
-            if ann.is_type(Uri.TOKEN) and ann.long_id.startswith(asr_vid):
-                sliced.append(ann.get('word'))
-        nv = mmif.new_view()
-        self.sign_view(nv, parameters)
-        ntd = nv.new_textdocument(' '.join(sliced))
-        return mmif
-                
-        
-        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._annotate
-
+        self.mmif = mmif if isinstance(mmif, Mmif) else Mmif(mmif)
+        self.text_doc = self.mmif.get_documents_by_type(DocumentTypes.TextDocument)
+        assert len(self.text_doc) == 1, "There should be exactly one TextDocument in the MMIF file"
+
+        labels = parameters["contain_labels"]
```

Comment:
if you set the parameter to be `multivalue=True' (https://sdk.clams.ai/appmetadata.html#runtime-parameter-specification) you don't need to do this "casting" yourself.

Reply:
(in that case, the default should be `[]` instead of `*`.